plugins/pvio/pvio_socket.c

```diff
@@ -786,7 +786,13 @@ my_bool pvio_socket_connect(MARIADB_PVIO *pvio, MA_PVIO_CINFO *cinfo)
     else
 #endif
     {
-      strcpy(UNIXaddr.sun_path, cinfo->unix_socket);
+      size_t sun_path_size = sizeof(UNIXaddr.sun_path);
+      strncpy(UNIXaddr.sun_path, cinfo->unix_socket, sun_path_size - 1);
+      if (sun_path_size == strlen(UNIXaddr.sun_path) + 1 && UNIXaddr.sun_path[sun_path_size - 1] != '\0')
+      {
+        // Making the string null-terminated
+        UNIXaddr.sun_path[sun_path_size -1] = '\0';
+      }
```

Comment:
A few lines above there's a check for `strlen(cinfo->unix_socket)`. If covscan gives false positives, why should we uglify the code? Same question re similar change in c95f86e88289ca84d881af6680e56c25aeb8e3b6.

Reply:
Yes you are right, I'll delete it and add covscan annotation

Reply:
Changed in a081d82e0c3e4b7a91915ed131a61dc4917c8e64